scipy/signal/_signaltools.py

```diff
@@ -1288,11 +1339,19 @@ def choose_conv_method(in1, in2, mode='full', measure=False):
     `convolve`.
 
     """
-    volume = np.asarray(in1)
-    kernel = np.asarray(in2)
+#    xp = array_namespace(in1, in2)
+    try:
+        xp = array_namespace(in1, in2)
+    except TypeError:
+        # either in1 or in2 are object arrays
+        xp = np_compat
+
+    if is_numpy(xp):
+        _reject_objects(in1, 'choose_conv_method')
+        _reject_objects(in1, 'choose_conv_method')
```

Comment:
above you had these `_reject_objects` lines in the `except TypeError` block.

Reply:
Good catch, thanks! Checking for objects in the `except TypeError:` block is wrong: if `in1` and `in2` are numpy object arrays, the exception is not raised and `_reject_arrays` are not called. Fixed now.